sources/Valkey.Glide/BaseClient.cs

```diff
@@ -38,6 +43,28 @@ public void Dispose()
 
     public override int GetHashCode() => (int)_clientPointer;
 
+    /// <summary>
+    /// Get the PubSub message queue for manual message retrieval.
+    /// Returns null if no PubSub subscriptions are configured.
+    /// Uses thread-safe access to prevent race conditions.
+    /// </summary>
+    public PubSubMessageQueue? PubSubQueue
+    {
+        get
+        {
+            lock (_pubSubLock)
+            {
+                return _pubSubHandler?.GetQueue();
+            }
+        }
+    }
+
+    /// <summary>
+    /// Indicates whether this client has PubSub subscriptions configured.
+    /// Uses volatile read for thread-safe access without locking.
+    /// </summary>
+    public bool HasPubSubSubscriptions => _pubSubHandler != null;
```

Comment:
Should this use same locking logic as `PubSubQueue` above to prevent race conditions, or just comment that it is possible for `HasPubSubSubscriptions` to return true when `PubSubQueue` is null?